### ISSUE TITLE
Serialize generated DDL

### ISSUE THREAD

**Issue by frioux** [frioux/DBIx-Class-DeploymentHandler#11]
Early on DBICDH actually generated an array of SQL serialized as JSON (1f0d0633940a59386e5c0ebec602d0650a3694ed).  For some dumb reason (ff7f0e7d4be69cb76ad7fc18ac22249a189126a5 ?) I removed the code (09bc35e3b405fbac725a45739e15cf30543d7f77).

This should be put back for goodness sakes!  It's much better, esp for databases like oracle where simply splitting on ; is insufficient.

**Comment by colinnewell** [comment]
This bug could do with more detail.  I could tell that it related to the SQL, but I don't understand what exactly you want to do.

Are you suggesting that we should provide json files with lists of statements?  I kind of read this or storing a json list of statements in the version table?

**Comment by frioux** [comment]
yeah when the SQL is generated it should store it as json or yaml or something, instead of a flat file of just sql with ";\n" joining the statements.

**Comment by colinnewell** [comment]
It sounds like we could do with,

1. the option to suck in json lists of SQL statements
2. store the instructions run as json.
3. improvements to the text file parsing, and possibly warnings (and pointers to json storage) if it looks like we might have misinterpreted the text file as per #47.

Does anything currently consume the instructions list from the version table?  Would that need to be an option?

**Comment by frioux** [comment]
nah the stuff in the table is just for auditing.  I think you're right re the options.  The one other thing to consider is legacy implementations.  If you turn on the flag, do we fall back from json to plaintext?  Do we offer a tool to upgrade plaintext to JSON?  etc.
